Automattic/wp-calypso
# JITMs: Remove per-site cache and per-page distinction for sidebar_notice

fixes #46809

#### Changes proposed in this Pull Request

* Remove the per-site JITM cache added in #46677
* Remove the ability for `:sidebar_notice` JITMs to be set differently per section.  Instead hardcode the messagePath to be the same, so pages like "Posts", "Pages", "Stats", etc all pull up the same JITMs from the left side.
* Remove the `messagePathSuffix` functionality from JITM.  It will only expect a basic `messagePath`.

#### Testing instructions



* Use calypso, both with `?flags=nav-unification` on and off
![2020-10-27_16-57](https://user-images.githubusercontent.com/937354/97366585-91208580-1875-11eb-888c-c491b44556bf.png)
* The pop-up notices should continue to work correctly and not momentarily disappear and reappear as you navigate through the site
* The `<SiteNotice>` component shouldn't re-render excessively
